File: examples/conditions/conditions.ts
import { deploy, out } from '../../helper/typescript/typescript_test';
import { Statement } from '../../lib';

function getStatement() {
  return (
    // doc-start
    new Statement.Ec2()
      .allow()
      .toStartInstances()
      .ifEncrypted()
      .ifInstanceType(['t3.micro', 't3.nano'])
      .ifAssociatePublicIpAddress(false)
      .ifAwsRequestTag('Owner', 'John')
    // doc-end
  );
}
const s = [getStatement()];
out(s);
deploy(s);

File: examples/conditions/conditions.py
import iam_floyd as statement
import importlib
import os
import sys
import inspect

currentdir = os.path.dirname(os.path.abspath(
    inspect.getfile(inspect.currentframe())))
helperDir = '%s/../../helper/python' % currentdir
sys.path.insert(0, helperDir)

test = importlib.import_module('python_test')
out = getattr(test, 'out')
deploy = getattr(test, 'deploy')

s = (
    # doc-start
    statement.Ec2() \
    .allow() \
    .to_start_instances() \
    .if_encrypted() \
    .if_instance_type(['t3.micro', 't3.nano']) \
    .if_associate_public_ip_address(False) \
    .if_aws_request_tag('Owner', 'John')
    # doc-end
)

all = [s]
out(all)
deploy(all)
